### PR TITLE
Make apps Qt6 compatible

### PR DESCRIPTION
**Pull request by mvieth**
Fixes https://github.com/PointCloudLibrary/pcl/issues/5518

### CODE REVIEW

File: apps/point_cloud_editor/CMakeLists.txt [hunk 1]
```diff
@@ -86,7 +86,11 @@ PCL_ADD_EXECUTABLE(
     ${RSRC}
     ${INCS})
 
-target_link_libraries("${EXE_NAME}" ${QTX}::Widgets ${QTX}::OpenGL ${OPENGL_LIBRARIES} ${BOOST_LIBRARIES} pcl_common pcl_io pcl_filters)
+if (${QTX} MATCHES "Qt6")
```
Comment (larshg) [inline apps/point_cloud_editor/CMakeLists.txt:89]: Alternatively we could do it like:
```
target_link_libraries(XXXX all_common_libs)
if (Qt Matches Qt6)
   target_link_libraries (XXXX ${QTX}::OpenGLWidgets)
endif()
```

File: apps/modeler/src/main_window.cpp [hunk 2]
```diff
@@ -282,7 +282,7 @@ pcl::modeler::MainWindow::updateRecentActions(
   }
 
   recent_items.removeDuplicates();
-  int recent_number = std::min(int(MAX_RECENT_NUMBER), recent_items.size());
+  int recent_number = std::min<int>(int(MAX_RECENT_NUMBER), recent_items.size());
```
Comment (larshg) [inline apps/modeler/src/main_window.cpp:285]: Why is it necessary to explicit tell std::min to use int?

Comment (mvieth) [reply apps/modeler/src/main_window.cpp:285]: In Qt6, recent_items.size() returns a qsizetype, which is usually not equal to int (more like ssize_t). So min doesn't know which type to use.

**Review by larshg — APPROVED** [review]

**Comment by themightyoarfish** [comment]
This PR breaks `pcl_point_cloud_editor`. 2D selection no longer works. Bisected the offending commit to `38cbd621d0632895e1bed04841a612f97f574abc`.

**Comment by larshg** [comment]
> This PR breaks `pcl_point_cloud_editor`. 2D selection no longer works. Bisected the offending commit to `38cbd621d0632895e1bed04841a612f97f574abc`.

With qt5 or 6?

**Comment by themightyoarfish** [comment]
This is qt 5.15.8 installed via `brew install qt@5`. I opened a corresponding issue here: https://github.com/PointCloudLibrary/pcl/issues/5680

**Pull request merged**